app/controllers/govuk_publishing_components/component_guide_controller.rb

```diff
@@ -48,14 +48,20 @@ def components_in_use_sass(print_styles)
       additional_files << "@import 'govuk_publishing_components/component_support';\n" unless print_styles
 
       components = components_in_use
-      extra_components = []
-
-      components.each do |component|
-        components_in_component = components_within_component(component)
-        extra_components << components_in_component
+      components_to_search = components
+      loop = true
+
+      while loop
```

Comment:
This looks more like a depth-first iterative search than a recursive one, so we might want to update the PR title and function name in the next commit.

Reply:
Have updated, okay to re-review?